uberfire-rest/uberfire-rest-backend/src/main/java/org/guvnor/rest/backend/ProjectResource.java

```diff
@@ -531,7 +485,13 @@ public Response deleteSpace(@PathParam("spaceName") String spaceName) {
         return createAcceptedStatusResponse(jobRequest);
     }
 
-    private Response createAcceptedStatusResponse(JobRequest jobRequest) {
+    private void assertObjectExists(final Object o) {
+        if (o == null) {
+            throw new WebApplicationException(Response.status(Status.NOT_FOUND).entity(o).build());
```

Comment:
Just wondering what's the use of adding `entity(o)` if it's always null. Wouldn't it be better to provide at least some string description so the "Not found" response is more useful?

Reply:
@Rikkola ping in case you missed my question in the flood of comments :-)

Reply:
Good point. Like a professional I just copy pasted this line in here. I'll take that out.

Reply:
@Rikkola Wouldn't it make more sense to add some info about what was null? How about adding String parameter to the `assertObjectExists(final Object o, String whatIsMissing)` and adding this to the exception somehow? Especially since some methods can fail with multiple "missing objects", like `deleteProject` project can fail because space is missing or project within the space is missing - informing user which was the case would be useful!

Reply:
Makes sense, I'll do this.